python/ray/serve/tests/test_failure.py

```diff
@@ -209,6 +210,41 @@ def __call__(self):
                 time.sleep(0.1)
 
 
+def test_create_backend_idempotent(serve_instance):
+    serve.init()
+
+    def f():
+        pass
+
+    controller = serve.api._get_controller()
+
+    replica_config = ReplicaConfig(f)
+    backend_config = BackendConfig({"num_replicas": 1})
+
+    for i in range(10):
+        controller.create_backend.remote("my_backend", backend_config,
+                                         replica_config)
+
+    assert len(ray.get(controller.get_all_backends.remote())) == 1
+
+
+def test_create_endpoint_idempotent(serve_instance):
+    serve.init()
+
+    def f():
+        pass
+
+    serve.create_backend("my_backend", f)
+
+    controller = serve.api._get_controller()
+
+    for i in range(10):
+        controller.create_endpoint.remote("my_endpoint", {"my_backend": 1.0},
+                                          "my_route", ["Get"])
+
+    assert len(ray.get(controller.get_all_endpoints.remote())) == 1
```

Comment:
Same here, maybe we can check the routes endpoint (`localhost:8000/-/routes`)

Reply:
Good idea, I'll add this.